org.osgi.test.junit5.cm/src/main/java/org/osgi/test/junit5/cm/ConfigurationCopy.java

```diff
@@ -0,0 +1,50 @@
+package org.osgi.test.junit5.cm;
+
+import java.util.Dictionary;
+
+import org.osgi.service.cm.Configuration;
+
+public class ConfigurationCopy {
+
+	private String						pid;
+	private Dictionary<String, Object>	properties;
+	private String						factoryPid;
+	private String						bundleLocation;
+	private long						changeCount;
+
+	private ConfigurationCopy() {
+
+	}
+
+	static ConfigurationCopy of(Configuration configuration) {
+
+		ConfigurationCopy copy = new ConfigurationCopy();
+		copy.pid = configuration.getPid();
+		copy.properties = configuration.getProperties();
```

Comment:
Shouldn't this be a deeper copy? Not just a reference?

Reply:
sure. will add a copy method into Dictionaries.class

Reply:
i copy the Dictionary. but maybe there is an better way to do this

https://github.com/osgi/osgi-test/pull/165/commits/4b57e835d598bdbdc72f515e916dbc74fd1f76be#diff-8abbc06a0596380453c03014d8e0ca12R448-R452